src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/NoWhitespaceBeforeCaseDefaultColonCheck.java

```diff
@@ -0,0 +1,220 @@
+ * </p>
+ * <ul>
+ * <li>
+ * {@code ws.preceded}
+ * </li>
+ * </ul>
+ *
+ * @since 8.44
+ */
+@StatelessCheck
+public class NoWhitespaceBeforeCaseDefaultColonCheck
+    extends AbstractCheck {
+
+    /**
+     * A key is pointing to the warning message text in "messages.properties"
+     * file.
+     */
+    public static final String MSG_KEY = "ws.preceded";
+
+    @Override
+    public int[] getDefaultTokens() {
+        return getRequiredTokens();
+    }
+
+    @Override
+    public int[] getAcceptableTokens() {
+        return getRequiredTokens();
+    }
+
+    @Override
+    public int[] getRequiredTokens() {
+        return new int[] {TokenTypes.COLON};
+    }
+
+    @Override
+    public boolean isCommentNodesRequired() {
+        return true;
+    }
+
+    @Override
+    public void visitToken(DetailAST ast) {
+        if (isInSwitch(ast) && isWhiteSpaceBeforeColon(ast)) {
+            log(ast, MSG_KEY, ast.getText());
+        }
+    }
+
+    /**
+     * Checks if the colon is inside a switch block.
+     *
+     * @param colonAst DetailAST to check.
+     * @return true, if colon case is inside a switch block.
+     */
+    private static boolean isInSwitch(DetailAST colonAst) {
+        return TokenUtil.isOfType(colonAst.getParent(), TokenTypes.LITERAL_CASE,
+                TokenTypes.LITERAL_DEFAULT);
+    }
+
+    /**
+     * Checks if there is a whitespace before the colon of a switch case or switch default.
+     *
+     * @param colonAst DetailAST to check.
+     * @return true, if there is whitespace preceding colonAst.
+     */
+    private static boolean isWhiteSpaceBeforeColon(DetailAST colonAst) {
+        final DetailAST parent = colonAst.getParent();
+        final boolean result;
+        if (isOnDifferentLineWithPreviousToken(colonAst)) {
+            result = true;
+        }
+        else if (parent.getType() == TokenTypes.LITERAL_CASE) {
+            result = isWhitespaceBeforeColonOfCase(colonAst);
+        }
+        else {
+            result = isWhitespaceBeforeColonOfDefault(colonAst);
+        }
+        return result;
+    }
+
+    /**
+     * Checks if there is a whitespace before the colon of a switch case.
+     *
+     * @param colonAst DetailAST to check.
+     * @return true, if there is whitespace preceding colonAst.
+     */
+    private static boolean isWhitespaceBeforeColonOfCase(DetailAST colonAst) {
+        final DetailAST previousSibling = colonAst.getPreviousSibling();
+        int offset = 0;
+        if (previousSibling.getType() == TokenTypes.BLOCK_COMMENT_BEGIN) {
+            offset = 1;
+        }
+        return colonAst.getColumnNo() != getLastColumnNumberOf(previousSibling) + offset;
+    }
+
+    /**
+     * Checks if there is a whitespace before the colon of a switch default.
+     *
+     * @param colonAst DetailAST to check.
+     * @return true, if there is whitespace preceding colonAst.
+     */
+    private static boolean isWhitespaceBeforeColonOfDefault(DetailAST colonAst) {
```

Comment:
`isWhitespaceBeforeColonOfCase` has custom support for comments. How come this isn't the same for default?

Reply:
Default also has comment support, only thing for `case` we need to specifically handle comment as there can be other expressions, but in case of `default`, since there can be only comment we are simply checking if there is any other element before colon
https://github.com/checkstyle/checkstyle/blob/8cce31164d76c5b4c527a163f6d94a3b4805a316/src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/NoWhitespaceBeforeCaseDefaultColonCheck.java#L176-L177